File: events/reset_password.py
# this sample Python code creates a Reset Password event that can be used to trigger a Reset Password flow.
# in a flow email triggered by this event, you can access the customer's unique password reset link using {{ event.PasswordResetLink }}
import os
from klaviyo_api import KlaviyoAPI

# load your private API key from environment variables. 
# ensure you have set the environment variable "private_api_key" with Klaviyo private API key that has write access to events. 
private_api_key = os.environ.get("private_api_key")

# instantiate the KlaviyoAPI client with your private API key.
klaviyo = KlaviyoAPI(private_api_key)

# replace the password reset link and customer email with your values.
password_reset_link = "https://www.yourwebsite.com/reset-password?unique_token=12345"
customer_email = "customer@email.com"

# create the Reset Password event.
body = {
    "data": {
        "type": "event",
        "attributes": {
            "properties": {
                "PasswordResetLink": password_reset_link
            },
            "metric": {
                "data": {
                    "type": "metric",
                    "attributes": {
                        "name": "Reset Password"
                    }
                }
            },
            "profile": {
                "data": {
                    "type": "profile",
                    "attributes": {
                        "email": customer_email
                    }
                }
            }
        }
    }
}

klaviyo.Events.create_event(body)

File: events/reset_password.js
// this sample JavaScript code creates a Reset Password event that can be used to trigger a Reset Password flow.
// in a flow email triggered by this event, you can access the customer's unique password reset link using {{ event.PasswordResetLink }}

import { ApiKeySession, EventsApi } from 'klaviyo-api'

// load private API key as an environment variable. Note that this API key needs to have write access to events.
// if using a .env file, ensure to load it using dotenv or similar package. 
const apiKey = process.env.KLAVIYO_API_KEY
if (!apiKey) {
  throw new Error('Please set the KLAVIYO_API_KEY environment variable.')
}

// instantiate the API session and events API
const session = new ApiKeySession(apiKey)
const eventsApi = new EventsApi(session)

const passwordResetLink = 'https://www.yourwebsite.com/reset-password?unique_token=12345'
const customerEmail = 'customer@email.com'

// set up event data details
const body = {
  data: {
    type: "event",
    attributes: {
      properties: {
        PasswordResetLink: passwordResetLink
      },
      metric: {
        data: {
          type: "metric",
          attributes: {
            name: "Reset Password"
          }
        }
      },
      profile: {
        data: {
          type: "profile",
          attributes: {
            email: customerEmail
          }
        }
      }
    }
  }
};

// create the event:
await eventsApi.createEvent(body);